torch_nets/test_python_net.py

```diff
@@ -1,26 +1,32 @@
-"""Module to save the results of running the net on a tensor of ones."""
+"""A smoke test for the ANN model.
 
-from torch import ones  # pylint: disable=no-name-in-module
-import models as nn_model
+This test checks that the model can be loaded from a weights file in both pt format and
+netcdf format and that they produce the expected output when given an input of all ones.
+This ensures that it is equivalent to the Fortran NN model.
+"""
 
+import os
+from pathlib import Path
+import torch
+import numpy as np
+from models import ANN, load_from_netcdf_params
 
-if __name__ == "__main__":
 
-    test_model = nn_model.ANN()
+os.chdir(Path(__file__).parent)
 
-    nn_model.endow_with_netcdf_params(
-        test_model,
-        "qobsTTFFFFFTF30FFTFTF30TTFTFTFFF80FFTFTTF2699FFFF_X01_no_qp_no_adv_surf_F"
-        + "_Tin_qin_disteq_O_Trad_rest_Tadv_qadv_qout_qsed_RESCALED"
-        + "_7epochs_no_drop_REAL_NN_layers5in61out148_BN_F_te70.nc",
-    )
+expected = np.loadtxt("nn_ones.txt").astype(np.float32)
 
-    test_model.eval()
+model1 = ANN().load("nn_state.pt")  # load from pt file
+model2 = load_from_netcdf_params(
+    "qobsTTFFFFFTF30FFTFTF30TTFTFTFFF80FFTFTTF2699FFFF_X01_no_qp_no_adv_"
+    "surf_F_Tin_qin_disteq_O_Trad_rest_Tadv_qadv_qout_qsed_RESCALED_7epochs"
+    "_no_drop_REAL_NN_layers5in61out148_BN_F_te70.nc"
+)  # load from netcdf file
 
-    output = test_model(ones(61))
+x = torch.ones(61)
 
-    print(output)
+actual1 = model1.forward(x).detach().numpy()
+actual2 = model2.forward(x).detach().numpy()
 
-    with open("nn_out.txt", "w", encoding="utf_8") as file:
-        for val in output:
-            file.write(f"{val.item()}\n")
+assert np.all(actual1 == actual2)
+assert np.allclose(expected, actual1, atol=1e-6, rtol=2e-6)
```

Comment:
Some documentation about this line would be good. Why this tolerance in particular for example?
I note that we discussed this tolerance varies on some machines and this is the tightest bound we could find. A quick comment to explain this would help future developers.

Reply:
I've updated `atol` to 3e-8 which is the lowest value that can pass the test on my machine. I've also added a comment here.